### PR TITLE
hard bitwise problem

### PR DESCRIPTION
**Pull request by envyaims**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

File: solutions/silver/cf-1088D.mdx [hunk 1]
```diff
@@ -0,0 +1,71 @@
+---
+id: cf-1088D
+source: CF
+title: Ehab and another another xor problem
+author: Chongtian Ma
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/63656)
```
Comment (ryanchou-dev) [inline solutions/silver/cf-1088D.mdx:8]:
```suggestion
[Official Editorial (C++)](https://codeforces.com/blog/entry/63656)
```

**Review by ryanchou-dev — APPROVED** [review]

File: solutions/silver/cf-1088D.mdx (hunk 1 above)
Comment (envyaims) [reply solutions/silver/cf-1088D.mdx:8]: there is no implementation...

File: solutions/silver/cf-1088D.mdx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+Here, we can query from the most significant to the least significant bit. This is because in bitwise, since higher bits always take
+priority, it is hard to bypass the effects of the higher bits if you don't already know them.
+
+First, we can submit a query $(c, d) = (0, 0)$ to figure out whether $a$ or $b$ is greater (this will become helpful later on). Now, loop from most to least significant bits. Let $\mathtt{curA}$ and $\mathtt{curB}$ be the values for the bits of $a$ and $b$ we already know. We can try to differentiate the two numbers at this bit. We can submit two queries, in the form $(\mathtt{curA}| 2^B, \mathtt{curB})$, and $(\mathtt{curA}, \mathtt{curB} | 2^B)$ where we are currently looping on bit $B$. We also know that there are the following four cases we have to consider:
+
+- both $a$ and $b$ has bit $B$ activated
+- neither $a$ and $b$ has bit $B$ activated
+- only $a$ has bit $B$ activated
+- only $b$ has bit $B$ activated
+
```
Comment (SansPapyrus683) [inline solutions/silver/cf-1088D.mdx:23]: capitalize the first letters of these

File: solutions/silver/cf-1088D.mdx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+int ask(int a, int b) {
+	cout << "? " << a << " " << b << endl;
+	int res;
+	cin >> res;
+	return res;
+}
+
+int main() {
+	bool a_greater_than_b = ask(0, 0) == 1;
+	int a = 0, b = 0;
```
Comment (SansPapyrus683) [inline solutions/silver/cf-1088D.mdx:47]: separate lines

File: solutions/silver/cf-1088D.mdx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	for (int bit = 29; bit >= 0; bit--) {
```
Comment (SansPapyrus683) [inline solutions/silver/cf-1088D.mdx:48]: not going to ask for a variable here
just explain where 29 comes from

File: solutions/silver/cf-1088D.mdx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+		} else {
+			if (a_greater_than_b) {
+				a |= 1 << bit;
+			} else {
+				b |= 1 << bit;
+			}
+			a_greater_than_b = query1 == 1;
+		}
+	}
+	cout << "! " << a << " " << b << endl;
```
Comment (SansPapyrus683) [inline solutions/silver/cf-1088D.mdx:65]:
```suggestion

	cout << "! " << a << " " << b << endl;
```

File: solutions/silver/cf-1088D.mdx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+	return res;
+}
+
+int main() {
+	bool a_greater_than_b = ask(0, 0) == 1;
+	int a = 0, b = 0;
+	for (int bit = 29; bit >= 0; bit--) {
+		int query1 = ask(a | (1 << bit), b);
+		int query2 = ask(a, b | (1 << bit));
+		if (query1 != query2) {
```
Comment (SansPapyrus683) [inline solutions/silver/cf-1088D.mdx:51]: add comments

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/silver/cf-1088D.mdx (hunk 3 above)
Comment (envyaims) [reply solutions/silver/cf-1088D.mdx:47]: why...

File: solutions/silver/cf-1088D.mdx (hunk 4 above)
Comment (envyaims) [reply solutions/silver/cf-1088D.mdx:48]: its in the explanation, the most significant bit

File: solutions/silver/cf-1088D.mdx (hunk 5 above)
Comment (envyaims) [reply solutions/silver/cf-1088D.mdx:65]: ok man

File: solutions/silver/cf-1088D.mdx (hunk 6 above)
Comment (envyaims) [reply solutions/silver/cf-1088D.mdx:51]: should be obvious from the explanation..

File: solutions/silver/cf-1088D.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/silver/cf-1088D.mdx:48]: i don't see the number 29 anywhere in the editorial

File: solutions/silver/cf-1088D.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/silver/cf-1088D.mdx:47]: wdym why
i've always asked for this

File: solutions/silver/cf-1088D.mdx (hunk 6 above)
Comment (SansPapyrus683) [reply solutions/silver/cf-1088D.mdx:51]: wouldn't hurt to clear things up a bit more

File: solutions/silver/cf-1088D.mdx (hunk 4 above)
Comment (envyaims) [reply solutions/silver/cf-1088D.mdx:48]: well i sure do hope they read the constraints..

File: solutions/silver/cf-1088D.mdx (hunk 3 above)
Comment (envyaims) [reply solutions/silver/cf-1088D.mdx:47]: but why

File: solutions/silver/cf-1088D.mdx (hunk 6 above)
Comment (envyaims) [reply solutions/silver/cf-1088D.mdx:51]: how
waht do i even say

File: solutions/silver/cf-1088D.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/silver/cf-1088D.mdx:47]: looks nicer + at this point it’s just for consistency

File: solutions/silver/cf-1088D.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/silver/cf-1088D.mdx:48]: no change = get dong/harry to merge

File: solutions/silver/cf-1088D.mdx (hunk 6 above)
Comment (SansPapyrus683) [reply solutions/silver/cf-1088D.mdx:51]: “handle case #x” here or smth
you wrote the code, you should know how to explain it

File: solutions/silver/cf-1088D.mdx (hunk 3 above)
Comment (envyaims) [reply solutions/silver/cf-1088D.mdx:47]: no ones gonna notice bruh

File: solutions/silver/cf-1088D.mdx (hunk 4 above)
Comment (envyaims) [reply solutions/silver/cf-1088D.mdx:49]: fine, u tyrant

**Comment by envyaims** [comment]
we really be babysitting the reader

File: solutions/silver/cf-1088D.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/silver/cf-1088D.mdx:49]: my guy i am literally just one pm
there’s like 4 others

**Comment by SansPapyrus683** [comment]
yes, that’s the point

File: solutions/silver/cf-1088D.mdx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+Here, we can query from the most significant to the least significant bit. This is because in bitwise, since higher bits always take
+priority, it is hard to bypass the effects of the higher bits if you don't already know them.
+
+First, we can submit a query $(c, d) = (0, 0)$ to figure out whether $a$ or $b$ is greater (this will become helpful later on). Now, loop from most to least significant bits. Let $\mathtt{curA}$ and $\mathtt{curB}$ be the values for the bits of $a$ and $b$ we already know. We can try to differentiate the two numbers at this bit. We can submit two queries, in the form $(\mathtt{curA}| 2^B, \mathtt{curB})$, and $(\mathtt{curA}, \mathtt{curB} | 2^B)$ where we are currently looping on bit $B$. We also know that there are the following four cases we have to consider:
+
+- Both $a$ and $b$ has bit $B$ activated
+- Neither $a$ and $b$ has bit $B$ activated
+- Only $a$ has bit $B$ activated
+- Only $b$ has bit $B$ activated
+
```
Comment (SansPapyrus683) [inline solutions/silver/cf-1088D.mdx:23]: if you refer to the cases with numbers make this a numbered list

**Review by Aadit-Ambadkar — APPROVED** [review]

File: solutions/silver/cf-1088D.mdx (hunk 7 above)
Comment (SansPapyrus683) [reply solutions/silver/cf-1088D.mdx:23]: unresolved

Comment (envyaims) [reply solutions/silver/cf-1088D.mdx]: on the live editor it's blue is it supposed to be like that lmao

Comment (SansPapyrus683) [reply solutions/silver/cf-1088D.mdx]: yeah

**Pull request merged**